### DISCUSSION TITLE
How can I have two status columns in the grid?

### DISCUSSION THREAD

**Discussion by gotzsys** [filamentphp/filament#6719] [Help] [answered]
I would like to have two status columns appearing in the grid, one with a SelectColumn so that I can update the status of the queue, and another with a BadgeColumn.

```
SelectColumn::make('status')
        ->label('Status 1')
        ->options(self::FILTER_STATUS)
        ->toggleable(),

BadgeColumn::make('status')
    ->label('Status')
    ->enum(self::FILTER_STATUS)
    ->colors([
        'primary' => 'waiting',
        'success' => 'seated',
        'danger' => static fn ($status): bool => in_array($status, ['called', 'canceled_by_client', 'canceled_by_company', 'no_show']),
    ])
    ->toggleable(),
```

**Comment by danharrin** [comment]
You cannot, columns must be unique. Why would you want both anyway?

**Reply by gotzsys** [reply, marked as answer]
Hello Dan, how are you?
An example would be a mobile field, where in the grid I would like to add this field twice.
1. The cell phone in text format.
2. A contact button that opens WhatsApp Web.

The data source would be the same, but with different purposes.
But anyway, I managed to solve it in a simple way. I created a dynamic attribute in the model:

```
public function getWhatsappAttribute()
{
     return $this->mobile ? preg_replace('/[^0-9]/', '', $this->mobile) : null;
}
```
